MonstieWash/Assets/Scripts/Player/Inventory.cs

```diff
@@ -0,0 +1,202 @@
+
+            //Update consumable ui when changing storage amount of them;
+            if ((type is Consumable)) RefreshConsumableUI();
+        }
+
+    }
+
+    /// <summary>
+    /// Removes a given item from storage.
+    /// </summary>
+    /// <param name="type">Type of item to remove</param>
+    public void RemoveItem(InventoryItem type)
+    {
+        var itemIndex = RetrieveItemIndex(type);
+        if (itemIndex != -1)
+        {
+            //Update consumable ui when changing storage amount of them;
+            if ((type is Consumable)) RefreshConsumableUI();
+            //Remove dictionary stored type.
+            storedItemsList.RemoveAt(itemIndex);
+        }
+        else Debug.LogWarning($"Item {type.ItemName} doesn't exist.");
+    }
+
+
+    /// <summary>
+    /// Takes a consumable and returns the index of it within this inventories list.
+    /// </summary>
+    /// <param name="i"></param>
+    /// <returns></returns>
+    public int RetrieveItemIndex(InventoryItem item)
+    {
+        for (int i = 0; i < storedItemsList.Count; i++)
+        {
+            if (storedItemsList[i].ItemData.ItemName == item.ItemName)
+            {
+                return i;
+            }
+        }
+        return -1;
+    }
+
+    /// <summary>
+    /// Simple function for other scripts to read the quanity of a given item;
+    /// </summary>
+    /// <param name="item">Item type to check for in inventory</param>
+    /// <returns></returns>
+    public int ReadQuantity(InventoryItem item)
+    {
+        var itemIndex = RetrieveItemIndex(item);
+        if (itemIndex == -1) return 0;
+        else return storedItemsList[itemIndex].Quantity;
+
+    }
+
+    /// <summary>
+    /// Simple function to check if inventory contains an item;
+    /// </summary>
+    /// <param name="item">Item type to check for in inventory</param>
+    /// <returns></returns>
+    public bool ContainsItem(InventoryItem item)
+    {
+        var itemIndex = RetrieveItemIndex(item);
+        if (itemIndex == -1) return false;
+        else return true;
+    }
+
+    /// <summary>
+    /// Updates the Bag ui at the bottom of the screen when called;
+    /// </summary>
+    /// <param name="type"></param>
+    public void RefreshConsumableUI()
+    {
+            ConsumablesManager cm;
+            if (cm = FindFirstObjectByType<ConsumablesManager>())
+            {
+            print("refreshing consumable ui from inv");
+                cm.RefreshUI();
+            }
+    }
+
+    /// <summary>
+    /// Assigns this object to not be destroyed on load. If a new script is found will then remove it;
+    /// </summary>
+    private void MakeClassStatic()
+    {
+        Inventory[] objs = FindObjectsByType<Inventory>(FindObjectsSortMode.None);
+
+        //When class already exists don't create a new script;
+        if (objs.Length > 1)
+        {
+            Destroy(this.gameObject);
+        }
+
+        DontDestroyOnLoad(this.gameObject);
```

Comment:
Just with this, adding it to DontDestroyOnLoad causes it to maintain its values even when the game is reset using the hotkey and we don't want that since we want every new player at pax to have a fresh start. Further, since the GameStartingScene is always active and will never be destroyed during play, as long as this script is part of that scene it will have the same effect as DontDestroyOnLoad anyways.

That's just a long way to say: this line should be removed.

```suggestion
        
```